extensions/infinispan-client/runtime/src/main/java/io/quarkus/infinispan/client/runtime/InfinispanTemplate.java

```diff
@@ -14,4 +16,11 @@ public BeanContainerListener configureInfinispan(Properties properties) {
             instance.configure(properties);
         };
     }
+
+    public void configureRuntimeProperties(InfinispanClientConfigRuntime infinispanClientConfigRuntime) {
+        InstanceHandle<InfinispanClientProducer> instance = Arc.container().instance(InfinispanClientProducer.class);
+        if (instance.isAvailable()) {
```

Comment:
@gsmet I changed here

Reply:
Hmmm, the bean is not initialized at all in this Camel test or we have a race condition and it's available later? (just checking)